html/inc/boinc_db.inc

```diff
@@ -57,31 +57,27 @@ class BoincDb extends DbConn {
             if ($host && $name && $user && $passwd) {
                 $retval = $instance->init_conn($user, $passwd, $host, $name);
                 if ($retval) {
-                    if ($instance->do_query("use $name")) {
-                        //error_log("BoincDb::get_aux(): connected to replica DB $dbnum");
-                        self::$instance = $instance;
-                        return;
-                    }
+                    //error_log("BoincDb::get_aux(): connected to replica DB $dbnum");
+                    self::$instance = $instance;
+                    return;
                 }
             }
             // if can't connect to replica, fall through and try DB 0
         }
         $host = parse_config($config, '<db_host>');
+        if (!$host) $host = 'localhost';
         $user = parse_config($config, '<db_user>');
         $name = parse_config($config, '<db_name>');
         $passwd = parse_config($config, '<db_passwd>');
-        // OK if host is null; it will use localhost
         if (!$name || !$user || !$passwd) {
             error_log("BoincDb::get_aux(): must specify DB name, user, passwd");
             return;
         }
         $retval = $instance->init_conn($user, $passwd, $host, $name);
         if ($retval) {
-            if ($instance->do_query("use $name")) {
-                //error_log("BoincDb::get_aux(): connected to DB $dbnum");
-                self::$instance = $instance;
-                return;
-            }
+            //error_log("BoincDb::get_aux(): connected to DB $dbnum");
```

Comment:
Please remove dead code like this. You can add it back for debugging if needed. Alternatively, use pre-processor macros.

Reply:
Fixed by @bema-aei